packages/site/src/components/DocWrapper.js

```diff
@@ -189,35 +221,29 @@ export default (props: Props) => {
         renderSidebar={({ setSidebarOpenState }) =>
           docList.map(item => {
             return (
-              <Route
-                path="/docs/:docName"
-                key={item.title}
-                render={({ match }) => {
-                  const { docName } = match.params
-                  return (
-                    <Box onClick={() => setSidebarOpenState(false)}>
-                      <h3
-                        className={cx({
-                          'docSearch-lvl0':
-                            docHeadingMap[docName] === item.title
-                        })}
-                      >
-                        {item.title}
-                      </h3>
-                      {item.items.map(slug => (
-                        <Link
-                          key={slug}
-                          className={linkStyles}
-                          activeClassName={cx(activeStyles, 'docSearch-lvl1')}
-                          to={`/docs/${slug}`}
-                        >
-                          {docMap[slug] || slug}
-                        </Link>
-                      ))}
-                    </Box>
-                  )
-                }}
-              />
+              <Switch>
+                <Route
+                  path="/docs/:docName"
+                  key={item.title}
+                  render={({ match }) => {
+                    const { docName } = match.params
+                    return renderSidebar(
+                      item,
+                      setSidebarOpenState,
+                      docMap,
+                      docName
+                    )
+                  }}
+                />
+                <Route
+                  exact
+                  path="/docs"
+                  key={item.title}
+                  render={() => {
+                    return renderSidebar(item, setSidebarOpenState, docMap)
```

Comment:
Could you make `renderSidebar` just a stateless component that you render here? I understand that using the component prop wouldn't work because you can't pass custom props. Sidebar could be a component to leverage React's diffing algorithm.